docs/src/migration.md

```diff
@@ -0,0 +1,112 @@
+    end
+end
+```
+Note that we reuse the `kalman_filter_slice` model in the `state_space_model` model. In the argument list of any `GraphPPL` model, we have to specify the Markov Blanket of the model we are defining. This means that all interfaces with the outside world have to be passed as arguments to the model. For the `kalman_filter_slice` model, we pass the previous state `prev_x`, the new state `new_x`, the observation `y` as well as the parameters `A`, `B`, `Q` and `P`. This means that, when invoking a submodel in a larger model, we can specify all components of the Markov Blanket. Note that, in the `state_space_model` model, we do not pass `y` as an argument to the `kalman_filter_slice` model. `GraphPPL` will infer that `y` is missing from the argument list and assign it to whatever is left of the `~` operator. Note that we also use the `new(x[i + 1])` syntax to create a new variable in the position of `x[i + 1]`. Since `y` is also passed in the argument list of the `state_space_model` model, we could have written this line with the equivalen statement `x[i + 1] ~ kalman_filter_slice(prev_x = x[i], y = y[i], A=A, B=B, Q=Q, P=P)`. However, to respect the generative direction of the model and to make the code more readable, we use the `new(x[i + 1])` syntax.
+## Constraint specification
+With the introduction of nested models, the specification of variational constraints becomes more difficult. In version `3.1`, variable names were uniquely defined in the model, which made it easy to specify constraints on variables. In version `4.0`, nested models can contain variables with the same name as their parents, even though they are distinct random variables. Therefore, we need to specify constraints on submodel level in the constraints macro. This is done with the `for q in _submodel_` syntax. For example:
+```julia
+@constraints begin
+    for q in kalman_filter_slice
+        q(new_x, prev_x, y) = q(new_x, prev_x)q(y)
+    end
+end
+```
+This pushes the same constraint to all instances of the `kalman_filter_slice` submodel. Of course, we'd like to have more flexibility in the constraints we can specify. Therefore, we can also specify constraints on a specific instance of the submodel. For example:
+```julia
+@constraints begin
+    for q in (kalman_filter_slice, 1)
+        q(new_x, prev_x, y) = q(new_x, prev_x)q(y)
+        q(new_x) :: Normal
+    end
+end
+```
+This pushes the constraint to the first instance of the `kalman_filter_slice` submodel. With this syntax, we can specify constraints on any instance of a submodel. 
+
+The function syntax for constraints is still supported. For example:
+```julia
+@constraints function ssm_constraints(factorize)
+    for q in kalman_filter_slice
+        if factorize
+            q(new_x, prev_x, y) = MeanField()
+        else
+            q(new_x, prev_x, y) = q(new_x, prev_x)q(y)
+        end
+    end
+end
+```
+
+## Meta specification
```

Comment:
We can note that `@meta` syntax is now more sophisticated and allows named tuple to add more properties, not only `meta`. This is important because now we cannot write 
```
x = randomvar() where { prod_constraint = ... }
```
in the model and it must be done via `@meta` instead.